cwms-data-api/src/main/java/cwms/cda/api/PropertyController.java

```diff
@@ -0,0 +1,296 @@
+            String acceptHeader = ctx.req.getContentType();
+            String formatHeader = acceptHeader != null ? acceptHeader : Formats.JSONV2;
+            ContentType contentType = Formats.parseHeader(formatHeader);
+            if (contentType == null) {
+                throw new FormattingException("Format header could not be parsed");
+            }
+            Property property = deserializeProperty(ctx.body(), contentType);
+            property.validate();
+            DSLContext dsl = getDslContext(ctx);
+            PropertyDao dao = new PropertyDao(dsl);
+            dao.updateProperty(property);
+            ctx.status(HttpServletResponse.SC_OK).json("Updated Property");
+        } catch (IOException ex) {
+            throw new IllegalArgumentException("Unable to parse property from content body", ex);
+        }
+
+    }
+
+    @OpenApi(
+            pathParams = {
+                @OpenApiParam(name = NAME, description = "Specifies the name of "
+                        + "the property to be deleted."),
+            },
+            queryParams = {
+                @OpenApiParam(name = OFFICE, description = "Specifies the owning office of "
+                        + "the property to be deleted."),
+                @OpenApiParam(name = CATEGORY_ID, description = "Specifies the category id of "
+                        + "the property to be deleted."),
+            },
+            description = "Delete CWMS Property",
+            method = HttpMethod.DELETE,
+            tags = {TAG},
+            responses = {
+                @OpenApiResponse(status = STATUS_204, description = "Property successfully deleted from CWMS."),
+                @OpenApiResponse(status = STATUS_404, description = "Based on the combination of "
+                        + "inputs provided the property was not found.")
+            }
+    )
+    @Override
+    public void delete(Context ctx, String name) {
+        String office = ctx.queryParam(OFFICE);
+        String category = ctx.queryParam(CATEGORY_ID);
+        try (Timer.Context ignored = markAndTime(DELETE)) {
+            DSLContext dsl = getDslContext(ctx);
+            PropertyDao dao = new PropertyDao(dsl);
+            dao.deleteProperty(office, category, name);
+            ctx.status(HttpServletResponse.SC_NO_CONTENT).json(name + " Deleted");
+        }
+    }
+
+    private static Property deserializeProperty(String body, ContentType contentType)
+            throws IOException {
+        ObjectMapper om = getObjectMapperForFormat(contentType);
+        Property retVal;
+        try {
+            retVal = om.readValue(body, Property.class);
+        } catch (Exception e) {
+            throw new IOException("Failed to deserialize property", e);
+        }
+        return retVal;
+    }
+
+    private static ObjectMapper getObjectMapperForFormat(ContentType contentType) {
+        ObjectMapper om;
+        if (ContentType.equivalent(Formats.JSONV2, contentType.toString())) {
```

Comment:
The content-type mapper was design so you can just pass the data type in and with an annotation on the datatype get the right formatter given the content-type string.

This function is duplication of all that logic.

Reply:
Looks like that was only done one-way. I implemented the deserialization and updated my code to use the Formats factory methods. I'll do a follow-up PR removing the redundant code that I had copied this logic from.